.github/workflows/build.yml

```diff
@@ -441,3 +441,42 @@ jobs:
             zetaclientd-ubuntu-20-amd64
 
 
+  discord_announcement:
+    #runs-on: ubuntu-latest
+    runs-on: ["zeta-runners"]
+    timeout-minutes: 15
+    if: startsWith(github.ref, 'refs/tags/')
+    needs:
+      - upgrade_path_testing
+  
+    steps:
+      - uses: actions/checkout@v3
+  
+      # - name: setup-git-credentials
+      #   uses: de-vri-es/setup-git-credentials@v2.0.8
+      #   with:
+      #     credentials: ${{ secrets.PAT_GITHUB_SERVICE_ACCT }}
```

Comment:
This code adds a new job that will run immediately after `upgrade_path_testing` job is completed but the creation of the upload and GitHub release won't have happened yet. This means you could post an announcement before the  release is ready. 

Creating a new job to run your action is good for testing but overkill for the final solution. You only need to add an announcement after the file upload has completed. You should be able to add it into an existing job